### PR TITLE
Extract package code `cli/cluster` to a separate library `lib/cluster`.

### PR DESCRIPTION
**Pull request by oleg-jukovec**
The patchset:

* Extracts separate libraries into a separate folder `tt/lib` instead of `tt/cli`.
* Moves logic for working with tcs/etcd and integrity checks into the `lib/cluster` from `tt-ee`.

### CODE REVIEW

**Comment by oleg-jukovec** [comment]
Path in the `tt-ee`: https://github.com/tarantool/tt-ee/pull/161

File: lib/cluster/integration_test.go [hunk 1]
```diff
@@ -0,0 +1,1059 @@
...
+	etcd, err := cluster.ConnectEtcd(cluster.EtcdOpts{Endpoints: endpoints})
+	require.NoError(t, err)
+	require.NotNil(t, etcd)
+	defer etcd.Close()
+
+	etcdPut(t, etcd, "/foo/config/", "foo")
+	etcdPut(t, etcd, "/foo/config/foo", "zoo")
+
+	data := []byte("zoo bar foo")
+	err = cluster.NewEtcdAllDataPublisher(etcd, "/foo/", timeout).Publish(0, data)
```
Comment (AlgebraicWolf) [inline lib/cluster/integration_test.go:527]: Don't we need to check this error?

File: lib/cluster/etcd.go [hunk 2]
```diff
@@ -0,0 +1,807 @@
...
+		ok, hash, key := parseHashPath(string(resp.Key), collector.prefix)
+		if ok && key == collector.key {
+			hashMap[hash] = resp.Value
+		}
+	}
+	sig := sigs[0].Value
+
+	err = collector.checkFunc(value, hashMap, sig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to make integration check for key %q: %w",
```
Comment (AlgebraicWolf) [inline lib/cluster/etcd.go:402]:
```suggestion
		return nil, fmt.Errorf("failed to perform integrity check for key %q: %w",
```

**Review by AlgebraicWolf — APPROVED** [review]
Thank you for the patch! Looks good to me w.r.t. the comments below

File: lib/cluster/etcd.go (hunk 2 above)
Comment (AlgebraicWolf) [reply lib/cluster/etcd.go]: "integration" != "integrity"
The two words have different meaning :(

**Review by psergee — APPROVED** [review]

File: .github/actions/prepare-ee-test-env/action.yml [hunk 3]
```diff
@@ -43,11 +51,13 @@ runs:
           key: ${{ matrix.sdk-version }}
 
       - name: Download Tarantool SDK
-        if: steps.cache-sdk.outputs.cache-hit != 'true'
         run: |
-          ARCHIVE_NAME=tarantool-enterprise-bundle-${{ matrix.sdk-version }}.tar.gz
+          ARCHIVE_NAME=tarantool-enterprise-sdk-${{ inputs.sdk-gc }}-${{ inputs.sdk-version }}.tar.gz
+          ARCHIVE_PATH=$(echo ${{ inputs.sdk-version }} | sed -rn \
+            's/^([0-9]+)\.([0-9]+)\.([0-9]+-){2}([a-z0-9]+-)?r[0-9]+\.([a-z]+)\.([a-z0-9_]+)$/${{ inputs.sdk-build }}\/\5\/\6\/\1\.\2/p')
+          echo download.tarantool.io/enterprise/${ARCHIVE_PATH}/${ARCHIVE_NAME}
```
[the remark this answers is not in the document]
Comment (oleg-jukovec) [reply .github/actions/prepare-ee-test-env/action.yml]: I'm sorry, debug artifacts - removed.

File: lib/integrity/integrity.go [hunk 4]
```diff
@@ -56,14 +56,16 @@ func InitializeIntegrityCheck(publicKeyPath string, configDir string) (Integrity
...
 	return nil, ErrNotConfigured
 }
 
-// NewDataPublisherFactory create a new DataPublisherFactory with integrity
-// algorithms in publishers. Should be never be called in the CE.
-func NewDataPublisherFactory(path string) (DataPublisherFactory, error) {
-	return nil, errors.New("integrity publishers should never be created in ce")
+// GetSignFunction returns a function that creates a map of hashes and a
+// signature for a data with the private key path.
+func GetSignFunction(path string) (
```
[the remark this answers is not in the document]
Comment (oleg-jukovec) [reply lib/integrity/integrity.go]: Updated: the arg name and the comment.

**Pull request merged**